src/Server/TCPHandler.cpp

```diff
@@ -962,10 +973,21 @@ bool TCPHandler::receivePacket()
     UInt64 packet_type = 0;
     readVarUInt(packet_type, *in);
 
-//    std::cerr << "Server got packet: " << Protocol::Client::toString(packet_type) << "\n";
 
     switch (packet_type)
     {
+        case Protocol::Client::NextTaskRequest:
+        {
+            auto id = receiveNextTaskRequest();
+#if USE_AWS_S3
+            auto next = TaskSupervisor::instance().getNextTaskForId(id);
+#else
+            auto next = "";
```

Comment:
I think we should throw here (and maybe send exception packet).
Sending empty string is a hiding of error.

Reply:
Deleted this code